### ISSUE TITLE
Recurring connect ETIMEDOUT errors with PnPjs in NodeJS when running in Azure functions

### ISSUE THREAD

**Issue by Yvand** [pnp/pnpjs#1878]
### Category
- [ ] Enhancement
- [ ] Bug
- [ ] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [2.8.0]

Please specify what version(s) of SharePoint you are targeting: [SharePoint Online]

### Question

I use PnPjs with a Typescript/NodeJS Azure function.
When I run it on my local machine it always works fine.
When I run in in Azure, many (but not all) of the update list item operations fail with this error:
`'FetchError: request to https://TENANT.sharepoint.com/teams/SITE/_vti_bin/client.svc failed, reason: connect ETIMEDOUT xx.xx.xx.xx:443'`

The function does many update operations, but each update is for a single list item.

**Comment by juliemturner** [comment]
Can you share how you're setting up the sp object?

**Comment by Yvand** [comment]
@juliemturner after many tests I was able to get rid of all the "connect ETIMEDOUT" errors.
I realized that perf is way much more limited in Azure Functions VS local executions:

Basically, I create many promises in parallel, and then wait for all of them to complete. Each of those promises is essentially doing a read/insert/update operation in SharePoint lists.
This was working like a charm in local executions, but in Azure it is generating many such "connect ETIMEDOUT" errors, probably due to some contention.
I fixed it by creating batches of 10 promises, wait for current batch to finish, and then starting next one.

Thank you for offering your help.

**Comment by koltyakov** [comment]
@Yvand yeah, in the cloud you create a bit different architecture with queues, states, logical retries, circuit breakers, etc. There is a log of possibilities but also unexpected bottlenecks. Sometime you'd prefer a simpler predictable and manageable data-flow. E.g. with high flow of actions to SharePoint API 429 and service unavailability should be expected not as something outstanding but what might happen time to time.
